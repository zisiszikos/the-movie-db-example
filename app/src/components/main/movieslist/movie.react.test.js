import React from 'react';
import Movie from './movie.react.js';
import nock from 'nock';
import configurationMock from 'test_data/configuration.json';
import moviesMock from 'test_data/movies.json';
import trailersMock from 'test_data/trailers.json';
import reviewsMock from 'test_data/reviews.json';

describe('Movie component', () => {

    beforeAll(() => {
        nock.disableNetConnect();
        nock('http://localhost:3000')
            .persist()
            .get(/\/api\/movies\/movie\/.*\/trailers/)
            .reply(200, trailersMock)
            .get(/\/api\/movies\/movie\/.*\/reviews/)
            .reply(200, reviewsMock)
            .get(/\/api\/movies\/movie\/.*\/similar/)
            .reply(200, moviesMock);
    });

    afterAll(() => {
        nock.cleanAll();
        nock.enableNetConnect();
    });

    test('should have correct initial values', async () => {
        const wrapper = mount(<Movie
            key={0}
            index={0}
            conf={configurationMock}
            movie={moviesMock[0]} />);

        await delayTest(1000);

        expect(wrapper.find('.movie').length).toEqual(1);
        expect(wrapper.find('.movie.expanded').length).toEqual(0);
        expect(wrapper.props().index).toEqual(0);
        expect(wrapper.props().movie).toBeInstanceOf(Object);
        expect(wrapper.props().movie).toMatchObject(moviesMock[0]);
        expect(wrapper.props().conf).toBeInstanceOf(Object);
        expect(wrapper.props().conf).toMatchObject(configurationMock);

        expect(wrapper.state().movieDetails).toBeNull();

        expect(wrapper.instance().getMovieIndex()).toEqual(0);

        expect(wrapper.instance().createBackdropImageUrl())
            .toEqual('http://image.tmdb.org/t/p/w780/wBzMnQ01R9w58W6ucltdYfOyP4j.jpg');

        expect(wrapper).toMatchSnapshot();

        wrapper.setProps({
            key: 1,
            index: 1,
            movie: {
                ...moviesMock[0],
                id: 2222,
                backdrop_path: null
            }
        });
        expect(wrapper.props().movie.id).toEqual(2222);
        expect(wrapper.instance().getMovieIndex()).toEqual(1);

        expect(wrapper).toMatchSnapshot();

        wrapper.unmount();
    });

    test('should expand movie', async () => {
        const wrapper = mount(<Movie
            key={0}
            index={0}
            conf={configurationMock}
            movie={moviesMock[0]} />);

        await delayTest(1000);

        wrapper.setProps({
            key: 1,
            index: 1,
            movie: {
                ...moviesMock[0],
                isExpanded: true
            }
        });

        await delayTest(1000);

        expect(wrapper.state().movieDetails).toBeInstanceOf(Object);
        expect(wrapper.state().movieDetails.trailers)
            .toEqual(expect.arrayContaining(trailersMock));
        expect(wrapper.state().movieDetails.reviews)
            .toEqual(expect.arrayContaining(reviewsMock));
        expect(wrapper.state().movieDetails.simiralMovies)
            .toEqual(expect.arrayContaining(moviesMock));

        expect(wrapper).toMatchSnapshot();

        wrapper.unmount();
    });
});
