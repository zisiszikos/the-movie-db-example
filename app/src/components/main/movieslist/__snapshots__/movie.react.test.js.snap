// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie component should expand movie 1`] = `
<Movie
  conf={
    Object {
      "change_keys": Array [
        "adult",
        "air_date",
        "also_known_as",
        "alternative_titles",
        "biography",
        "birthday",
        "budget",
        "cast",
        "certifications",
        "character_names",
        "created_by",
        "crew",
        "deathday",
        "episode",
        "episode_number",
        "episode_run_time",
        "freebase_id",
        "freebase_mid",
        "general",
        "genres",
        "guest_stars",
        "homepage",
        "images",
        "imdb_id",
        "languages",
        "name",
        "network",
        "origin_country",
        "original_name",
        "original_title",
        "overview",
        "parts",
        "place_of_birth",
        "plot_keywords",
        "production_code",
        "production_companies",
        "production_countries",
        "releases",
        "revenue",
        "runtime",
        "season",
        "season_number",
        "season_regular",
        "spoken_languages",
        "status",
        "tagline",
        "title",
        "translations",
        "tvdb_id",
        "tvrage_id",
        "type",
        "video",
        "videos",
      ],
      "images": Object {
        "backdrop_sizes": Array [
          "w300",
          "w780",
          "w1280",
          "original",
        ],
        "base_url": "http://image.tmdb.org/t/p/",
        "logo_sizes": Array [
          "w45",
          "w92",
          "w154",
          "w185",
          "w300",
          "w500",
          "original",
        ],
        "poster_sizes": Array [
          "w92",
          "w154",
          "w185",
          "w342",
          "w500",
          "w780",
          "original",
        ],
        "profile_sizes": Array [
          "w45",
          "w185",
          "h632",
          "original",
        ],
        "secure_base_url": "https://image.tmdb.org/t/p/",
        "still_sizes": Array [
          "w92",
          "w185",
          "w300",
          "original",
        ],
      },
    }
  }
  index={1}
  movie={
    Object {
      "adult": false,
      "backdrop_path": "/wBzMnQ01R9w58W6ucltdYfOyP4j.jpg",
      "genre_ids": Array [
        28,
        12,
        35,
        14,
        878,
      ],
      "genre_names": Array [
        "Thriller",
        "Horror",
        "Drama",
      ],
      "id": 284053,
      "isExpanded": true,
      "original_language": "en",
      "original_title": "Thor: Ragnarok",
      "overview": "Thor is imprisoned on the other side of the universe and finds himself in a race against time to get back to Asgard to stop Ragnarok, the destruction of his homeworld and the end of Asgardian civilization, at the hands of an all-powerful new threat, the ruthless Hela.",
      "popularity": 793.613813,
      "poster_path": "/oSLd5GYGsiGgzDPKTwQh7wamO8t.jpg",
      "release_date": "2017-10-25",
      "title": "Thor: Ragnarok",
      "video": false,
      "vote_average": 7.6,
      "vote_count": 1617,
    }
  }
>
  <div
    className="movie expanded"
    data-index={1}
    onClick={[Function]}
    style={
      Object {
        "order": -1,
      }
    }
  >
    <div
      className="movie-box"
      style={
        Object {
          "backgroundImage": "url(http://image.tmdb.org/t/p/w780/wBzMnQ01R9w58W6ucltdYfOyP4j.jpg)",
        }
      }
    >
      <div
        className="tint"
      />
      <div
        className="movie-score"
      >
        7.6
      </div>
      <h3
        className="movie-title"
      >
        Thor: Ragnarok
         (
        2017
        )
      </h3>
      <p
        className="movie-genres"
      >
        Thriller, Horror, Drama
      </p>
      <p
        className="movie-overview"
      >
        Thor is imprisoned on the other side of the universe and finds himself in a race against time to get back to Asgard to stop Ragnarok, the destruction of his homeworld and the end of Asgardian civilization, at the hands of an all-powerful new threat, the ruthless Hela.
      </p>
    </div>
  </div>
</Movie>
`;

exports[`Movie component should have correct initial values 1`] = `
<Movie
  conf={
    Object {
      "change_keys": Array [
        "adult",
        "air_date",
        "also_known_as",
        "alternative_titles",
        "biography",
        "birthday",
        "budget",
        "cast",
        "certifications",
        "character_names",
        "created_by",
        "crew",
        "deathday",
        "episode",
        "episode_number",
        "episode_run_time",
        "freebase_id",
        "freebase_mid",
        "general",
        "genres",
        "guest_stars",
        "homepage",
        "images",
        "imdb_id",
        "languages",
        "name",
        "network",
        "origin_country",
        "original_name",
        "original_title",
        "overview",
        "parts",
        "place_of_birth",
        "plot_keywords",
        "production_code",
        "production_companies",
        "production_countries",
        "releases",
        "revenue",
        "runtime",
        "season",
        "season_number",
        "season_regular",
        "spoken_languages",
        "status",
        "tagline",
        "title",
        "translations",
        "tvdb_id",
        "tvrage_id",
        "type",
        "video",
        "videos",
      ],
      "images": Object {
        "backdrop_sizes": Array [
          "w300",
          "w780",
          "w1280",
          "original",
        ],
        "base_url": "http://image.tmdb.org/t/p/",
        "logo_sizes": Array [
          "w45",
          "w92",
          "w154",
          "w185",
          "w300",
          "w500",
          "original",
        ],
        "poster_sizes": Array [
          "w92",
          "w154",
          "w185",
          "w342",
          "w500",
          "w780",
          "original",
        ],
        "profile_sizes": Array [
          "w45",
          "w185",
          "h632",
          "original",
        ],
        "secure_base_url": "https://image.tmdb.org/t/p/",
        "still_sizes": Array [
          "w92",
          "w185",
          "w300",
          "original",
        ],
      },
    }
  }
  index={0}
  movie={
    Object {
      "adult": false,
      "backdrop_path": "/wBzMnQ01R9w58W6ucltdYfOyP4j.jpg",
      "genre_ids": Array [
        28,
        12,
        35,
        14,
        878,
      ],
      "genre_names": Array [
        "Thriller",
        "Horror",
        "Drama",
      ],
      "id": 284053,
      "original_language": "en",
      "original_title": "Thor: Ragnarok",
      "overview": "Thor is imprisoned on the other side of the universe and finds himself in a race against time to get back to Asgard to stop Ragnarok, the destruction of his homeworld and the end of Asgardian civilization, at the hands of an all-powerful new threat, the ruthless Hela.",
      "popularity": 793.613813,
      "poster_path": "/oSLd5GYGsiGgzDPKTwQh7wamO8t.jpg",
      "release_date": "2017-10-25",
      "title": "Thor: Ragnarok",
      "video": false,
      "vote_average": 7.6,
      "vote_count": 1617,
    }
  }
>
  <div
    className="movie "
    data-index={0}
    onClick={[Function]}
    style={
      Object {
        "order": 0,
      }
    }
  >
    <div
      className="movie-box"
      style={
        Object {
          "backgroundImage": "url(http://image.tmdb.org/t/p/w780/wBzMnQ01R9w58W6ucltdYfOyP4j.jpg)",
        }
      }
    >
      <div
        className="tint"
      />
      <div
        className="movie-score"
      >
        7.6
      </div>
      <h3
        className="movie-title"
      >
        Thor: Ragnarok
         (
        2017
        )
      </h3>
      <p
        className="movie-genres"
      >
        Thriller, Horror, Drama
      </p>
      <p
        className="movie-overview"
      >
        Thor is imprisoned on the other side of the universe and finds himself in a race against time to get back to Asgard to stop Ragnarok, the destruction of his homeworld and the end of Asgardian civilization, at the hands of an all-powerful new threat, the ruthless Hela.
      </p>
    </div>
  </div>
</Movie>
`;

exports[`Movie component should have correct initial values 2`] = `
<Movie
  conf={
    Object {
      "change_keys": Array [
        "adult",
        "air_date",
        "also_known_as",
        "alternative_titles",
        "biography",
        "birthday",
        "budget",
        "cast",
        "certifications",
        "character_names",
        "created_by",
        "crew",
        "deathday",
        "episode",
        "episode_number",
        "episode_run_time",
        "freebase_id",
        "freebase_mid",
        "general",
        "genres",
        "guest_stars",
        "homepage",
        "images",
        "imdb_id",
        "languages",
        "name",
        "network",
        "origin_country",
        "original_name",
        "original_title",
        "overview",
        "parts",
        "place_of_birth",
        "plot_keywords",
        "production_code",
        "production_companies",
        "production_countries",
        "releases",
        "revenue",
        "runtime",
        "season",
        "season_number",
        "season_regular",
        "spoken_languages",
        "status",
        "tagline",
        "title",
        "translations",
        "tvdb_id",
        "tvrage_id",
        "type",
        "video",
        "videos",
      ],
      "images": Object {
        "backdrop_sizes": Array [
          "w300",
          "w780",
          "w1280",
          "original",
        ],
        "base_url": "http://image.tmdb.org/t/p/",
        "logo_sizes": Array [
          "w45",
          "w92",
          "w154",
          "w185",
          "w300",
          "w500",
          "original",
        ],
        "poster_sizes": Array [
          "w92",
          "w154",
          "w185",
          "w342",
          "w500",
          "w780",
          "original",
        ],
        "profile_sizes": Array [
          "w45",
          "w185",
          "h632",
          "original",
        ],
        "secure_base_url": "https://image.tmdb.org/t/p/",
        "still_sizes": Array [
          "w92",
          "w185",
          "w300",
          "original",
        ],
      },
    }
  }
  index={1}
  movie={
    Object {
      "adult": false,
      "backdrop_path": null,
      "genre_ids": Array [
        28,
        12,
        35,
        14,
        878,
      ],
      "genre_names": Array [
        "Thriller",
        "Horror",
        "Drama",
      ],
      "id": 2222,
      "original_language": "en",
      "original_title": "Thor: Ragnarok",
      "overview": "Thor is imprisoned on the other side of the universe and finds himself in a race against time to get back to Asgard to stop Ragnarok, the destruction of his homeworld and the end of Asgardian civilization, at the hands of an all-powerful new threat, the ruthless Hela.",
      "popularity": 793.613813,
      "poster_path": "/oSLd5GYGsiGgzDPKTwQh7wamO8t.jpg",
      "release_date": "2017-10-25",
      "title": "Thor: Ragnarok",
      "video": false,
      "vote_average": 7.6,
      "vote_count": 1617,
    }
  }
>
  <div
    className="movie "
    data-index={1}
    onClick={[Function]}
    style={
      Object {
        "order": 1,
      }
    }
  >
    <div
      className="movie-box"
      style={
        Object {
          "backgroundImage": "url(/movie-default.jpg)",
        }
      }
    >
      <div
        className="tint"
      />
      <div
        className="movie-score"
      >
        7.6
      </div>
      <h3
        className="movie-title"
      >
        Thor: Ragnarok
         (
        2017
        )
      </h3>
      <p
        className="movie-genres"
      >
        Thriller, Horror, Drama
      </p>
      <p
        className="movie-overview"
      >
        Thor is imprisoned on the other side of the universe and finds himself in a race against time to get back to Asgard to stop Ragnarok, the destruction of his homeworld and the end of Asgardian civilization, at the hands of an all-powerful new threat, the ruthless Hela.
      </p>
    </div>
  </div>
</Movie>
`;
